src/controllers/csi/gc/controller.go

```diff
@@ -132,17 +142,31 @@ func collectGCInfo(ctx context.Context, apiReader client.Reader, dynakube dynatr
 	}, nil
 }
 
+func isSafeToGC(access metadata.Access, dynakubeList *dynatracev1beta1.DynaKubeList) bool {
+	dkMetadataList, err := access.GetAllDynakubes()
+	if err != nil {
+		log.Info("failed to get dynakube metadata from database, err: %s")
+		return false
+	}
+	filteredDynakubes := filterCodeModulesImageDynakubes(dynakubeList)
+	for _, dkMetadata := range dkMetadataList {
+		if dkMetadata.LatestVersion == "" {
+			return false
+		}
+		dynakube, ok := filteredDynakubes[dkMetadata.Name]
```

Comment:
```suggestion
		dynakube, hasDynakube := filteredDynakubes[dkMetadata.Name]
```

Reply:
Imho `ok` is better because it's standard way of checking it item exists in a map.

Reply:
I moved this part into a separate (2 line) function and in that case `ok` is better IMO